src/document.rs

```diff
@@ -39,18 +39,71 @@ fn read_file<P: AsRef<Path>>(path: P) -> Result<String> {
     Ok(text)
 }
 
+fn yaml_btreemap_to_hashmap(input: &BTreeMap<Yaml, Yaml>) -> HashMap<String, Value> {
```

Comment:
Preface: I'm still experimenting with Rust by pushing the language conventions to the limit so I can learn their strengths and weaknesses.  This might actually be a bad idea :)

You might not even need a function here.  In my experimenting with TOML support (mostly for debug dumps), I found doing the conversion the "functional" way felt clean to me (inmates running the asylum?).  Longer term, we should probably move this stuff to liquid-rs itself and take advantage of serde to do so.

For an idea of how this might look, here is my toml code:
```rust
        liquid::Value::Object(ref t) => {
            let toml_object: Result<BTreeMap<String, toml::Value>> = t.iter()
                .map(|(k, v)| convert_value(v).map(|v| (k.to_string(), v)))
                .collect();
            let toml_object = toml_object?;
            Ok(toml::Value::Table(toml_object))
        }
```

A simpler way for handling yaml's flexible keys is
```rust
yaml_key.as_str().expect("Keys must be strings").to_owned()
```

Ideally, we'd switch this function to returning a Result so we could avoid the panic/expect.